### PR TITLE
7126 but different

### PR DESCRIPTION
**Pull request by gmarinov**
FAO @Bobkidbob 

note base #1150

### CODE REVIEW

**Comment by gmarinov** [comment]
@benolot this goes into #1150 if @Bobkidbob likes it enough

File: src/Controller/Helpers/Breadcrumbs.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+class Breadcrumbs
+{
+    /* @var Breadcrumb[] */
+    protected $breadcrumbs = [];
+
+    /* @var BreadcrumbHelper */
+    protected $helper;
+
+    public function __construct(BreadcrumbHelper $helper, $initial = [])
```
Comment (Bobkidbob) [inline src/Controller/Helpers/Breadcrumbs.php]:
```suggestion
    public function __construct(BreadcrumbHelper $helper, array $initial = [])
```

File: src/Controller/Helpers/Breadcrumbs.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    {
+        $this->helper = $helper;
+        $this->breadcrumbs = [] + $initial;
```
Comment (Bobkidbob) [inline src/Controller/Helpers/Breadcrumbs.php]:
```suggestion
        $this->breadcrumbs = $initial;
```

File: src/Controller/Helpers/Breadcrumbs.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        );
+    }
+
+    public function forNetwork($network)
+    {
+        $this->breadcrumbs[] = $this->helper->getNetworkBreadcrumb($network);
+        return $this;
+    }
+
+    public function forRoute($title, $routeName, $parameters = [])
```
Comment (Bobkidbob) [inline src/Controller/Helpers/Breadcrumbs.php]:
```suggestion
    public function forRoute(string $title, string $routeName, array $parameters = []): Breadcrumbs
```

File: src/Controller/Helpers/Breadcrumbs.php [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            $isiteObj->getTitle(),
+            $routeName,
+            [
+                'key' => $isiteObj->getKey(),
+                'slug' => $isiteObj->getSlug(),
+            ]
+        );
+    }
+
+    public function forNetwork($network)
```
Comment (Bobkidbob) [inline src/Controller/Helpers/Breadcrumbs.php]:
```suggestion
    public function forNetwork(Network $network): Breadcrumbs
```

File: src/Controller/Helpers/Breadcrumbs.php [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        $this->breadcrumbs = [] + $initial;
+    }
+
+    public function forEntityAncestry($entity)
+    {
+        array_push($this->breadcrumbs, ...$this->helper->getCoreEntityAncestryBreadcrumbs($entity));
+        return $this;
+    }
+
+    public function forCategoryAncestry($category)
```
Comment (Bobkidbob) [inline src/Controller/Helpers/Breadcrumbs.php]:
```suggestion
    public function forCategoryAncestry(Category $category): Breadcrumbs
```

File: src/Controller/Helpers/Breadcrumbs.php [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    /* @var BreadcrumbHelper */
+    protected $helper;
+
+    public function __construct(BreadcrumbHelper $helper, $initial = [])
+    {
+        $this->helper = $helper;
+        $this->breadcrumbs = [] + $initial;
+    }
+
+    public function forEntityAncestry($entity)
```
Comment (Bobkidbob) [inline src/Controller/Helpers/Breadcrumbs.php]:
```suggestion
    public function forEntityAncestry(CoreEntity $entity): Breadcrumbs
```

File: src/Controller/Helpers/Breadcrumbs.php [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        return $this;
+    }
+
+    public function forCategoryAncestry($category)
+    {
+        array_push($this->breadcrumbs, ...$this->helper->getCategoryAncestryBreadcrumbs($category));
+        return $this;
+    }
+
+    public function forIsiteRoute(BaseIsiteObject $isiteObj, $routeName)
```
Comment (Bobkidbob) [inline src/Controller/Helpers/Breadcrumbs.php]:
```suggestion
    public function forIsiteRoute(BaseIsiteObject $isiteObj, string $routeName): Breadcrumbs
```

File: src/Controller/Helpers/Breadcrumbs.php [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        return $this;
+    }
+
+    public function clear()
+    {
+        $this->breadcrumbs = [];
+        return $this;
+    }
+
+    public function toArray()
```
Comment (Bobkidbob) [inline src/Controller/Helpers/Breadcrumbs.php]:
```suggestion
    public function toArray(): array
```

File: src/Controller/Helpers/Breadcrumbs.php [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+        return $this;
+    }
+
+    public function forRoute($title, $routeName, $parameters = [])
+    {
+        $this->breadcrumbs[] = $this->helper->createBreadcrumbForRoute($title, $routeName, $parameters);
+        return $this;
+    }
+
+    public function clear()
```
Comment (Bobkidbob) [inline src/Controller/Helpers/Breadcrumbs.php]: Is the `clear()` function needed?

Comment (Bobkidbob) [reply src/Controller/Helpers/Breadcrumbs.php]: If it is, should probably add a type hint.
```suggestion
    public function clear(): Breadcrumbs
```

File: src/Controller/Helpers/Breadcrumbs.php [hunk 10]
```diff
@@ -0,0 +1,69 @@
+<?php
+declare(strict_types = 1);
+
+namespace App\Controller\Helpers;
+
+use App\ExternalApi\Isite\Domain\BaseIsiteObject;
+use App\ValueObject\Breadcrumb;
```
Comment (Bobkidbob) [inline src/Controller/Helpers/Breadcrumbs.php]:
```suggestion
use App\ValueObject\Breadcrumb;
use BBC\ProgrammesPagesService\Domain\Entity\Category;
use BBC\ProgrammesPagesService\Domain\Entity\CoreEntity;
use BBC\ProgrammesPagesService\Domain\Entity\Network;
```

File: src/Controller/Recipes/RecipesController.php [hunk 11]
```diff
@@ -1,6 +1,7 @@
 <?php
 namespace App\Controller\Recipes;
 
+use App\Controller\Helpers\Breadcrumbs;
```
Comment (Bobkidbob) [inline src/Controller/Recipes/RecipesController.php]: Is this needed?
```suggestion
```

**Pull request merged**